src/utils/restUtilMod.F90.in

```diff
@@ -384,6 +387,7 @@ contains
     integer          , intent(in), optional :: lowerb2
     integer          , intent(in), optional :: upperb2 
     character(len=*) , intent(in), optional :: units            ! long name for variable
+    logical          , intent(in), optional :: scale_by_thickness  ! to scale by thickness in vertical interpolation or not
```

Comment:
On the first point, I will leave as is for now as suggested, and we will revisit before the end of the PR.

Working on the second point made me think further about what we're trying to do:

I've come up with this paragraph for now...
```
  ! iflag_no_scale_by_thickness => init_interp WILL NOT scale these soil
  !   variables by soil layer thickness. This is the default option.
  ! iflag_scale_by_thickness => init_interp WILL scale these soil variables
  !   by soil layer thickness to interpolate correctly from
  !   the soil profile present in the initial/restart file to the soil profile
  !   used by the model. For an example of setting this option to .true.,
  !   see H2OSOI_LIQ in subroutine Restart in WaterStateType.F90.
  !   Soil variables of so called extensive properties require this
  !   scaling for correct interpolation. Extensive properties depend on the
  !   sample size. H2OSOI_LIQ with units of kg/m2 contains fewer kg of
  !   water in thinner layers than in thicker layers.
  !   Scaling the quantity by layer thickness contributes the correct amount
  !   of water by adding rather than averaging amounts during interpolation.
  !   Not scaling by thickness results in underestimated soil water when
  !   interpolating from a finer to a coarser soil profile and
  !   overestimated soil water in the opposite case.
```
I mention "adding" versus "averaging" here, and I now think **that's** the distinction that we need to make between variables. Extensive properties need weighted summation, while intensive properties need weighted averaging. In other words, I think that even the temperature variables will benefit from weighting by soil layer thicknesses.

Pls let me know if you think that I have confused myself and have gone off on the wrong track.

Reply:
Your comment is great – thanks.

I can convince myself either way in terms of temperature and other, similar variables. I have an easier time making judgment calls about things like this if I set up a really extreme example. So I'll do that here, largely for my own benefit in thinking this through:

Let's consider two layers in the source grid: a top layer with dz = 0.01 m (midpoint at 0.005 m) and temperature = 280K, and a lower layer with dz = 0.99 m (midpoint at 0.505 m) and temperature = 300 K. Let's assume the destination has a top layer with dz = 0.51 m (midpoint at 0.255 m). Note that I have constructed this so that the destination layer's midpoint lies exactly half-way between the layer midpoints of the two source layers.

With the way the interpolation currently happens (unweighted), the destination temperature would be the mean of the two source layers, or 290K. I think that this amounts to assuming that the actual temperature varies linearly between the nodal values, which feels like a reasonable assumption to me. If we instead weighted each value by the thickness of the layer, then we'd end up with a destination temperature very close to that of the much larger, lower layer: 300K. I'm not really sure what that would imply about our assumptions for the temperature profile within a layer, but I guess it would be some nonlinear profile?

**Thinking through this makes me inclined to stick with the current, simpler linear interpolation for these quantities, but I could be convinced otherwise.**

Reply:
Removed "optional" from scale_by_thickness.